ddtrace/contrib/internal/anthropic/utils.py

```diff
@@ -68,30 +69,30 @@ def tag_tool_result_input_on_span(integration, span, chat_input, message_idx, bl
             span.set_tag_str(
                 "anthropic.request.messages.%d.content.%d.tool_result.content.%d.type"
                 % (message_idx, block_idx, tool_block_idx),
-                tool_block_type,
+                str(tool_block_type),
             )
 
 
 def tag_tool_use_output_on_span(integration, span, chat_completion, idx):
     tool_name = _get_attr(chat_completion, "name", None)
     tool_inputs = _get_attr(chat_completion, "input", None)
     if tool_name:
-        span.set_tag_str("anthropic.response.completions.content.%d.tool_call.name" % (idx), tool_name)
+        span.set_tag_str("anthropic.response.completions.content.%d.tool_call.name" % idx, str(tool_name))
     if tool_inputs:
         span.set_tag_str(
-            "anthropic.response.completions.content.%d.tool_call.input" % (idx),
-            integration.trunc(json.dumps(tool_inputs)),
+            "anthropic.response.completions.content.%d.tool_call.input" % idx,
+            integration.trunc(json.dumps(tool_inputs, default=_unserializable_default_repr)),
         )
 
 
 def tag_params_on_span(span, kwargs, integration):
     tagged_params = {}
     for k, v in kwargs.items():
         if k == "system" and integration.is_pc_sampled_span(span):
-            span.set_tag_str("anthropic.request.system", integration.trunc(v))
+            span.set_tag_str("anthropic.request.system", integration.trunc(str(v)))
         elif k not in ("messages", "model"):
             tagged_params[k] = v
-    span.set_tag_str("anthropic.request.parameters", json.dumps(tagged_params))
+    span.set_tag_str("anthropic.request.parameters", json.dumps(tagged_params, default=_unserializable_default_repr))
```

Comment:
per @lievan's comment offline, I also think it would be a good idea to have some kind of regression test specifically for the case of unserializable objects.